lampod/src/jsonrpc/channels.rs

```diff
@@ -8,3 +16,87 @@ pub fn json_list_channels(ctx: &LampoDeamon, request: &json::Value) -> Result<js
     let resp = ctx.channel_manager().list_channel();
     Ok(json::to_value(resp)?)
 }
+
+pub fn json_close_channel(ctx: &LampoDeamon, request: &json::Value) -> Result<json::Value, Error> {
+    log::info!("call for `closechannel` with request {:?}", request);
+    let mut request: request::CloseChannel = json::from_value(request.clone())?;
+    let events = ctx.handler().events();
+    // This gives all the channels with associated peer
+    let channels: response::Channels = ctx
+        .handler()
+        .call(
+            "channels",
+            json::json!({
+            "peer_id": request.node_id,
+            }),
+        )
+        .map_err(|err| {
+            Error::Rpc(RpcError {
+                code: -1,
+                message: format!("{err}"),
+                data: None,
+            })
+        })?;
+    let res = if channels.channels.len() > 1 {
+        // check the channel_id if it is not none, if it is return an error
+        // and if it is not none then we need to have the channel_id that needs to be shut
+        if request.channel_id.is_none() {
+            return Err(Error::Rpc(RpcError {
+                code: -1,
+                message: format!("Channels > 1, provide `channel_id`"),
+                data: None,
+            }));
+        } else {
+            // ctx.channel_manager().close_channel(request.clone());
+            request.clone()
+        }
+    } else if !channels.channels.is_empty() {
+        // This is the case where channel with the given node_id = 1
+        // SAFETY: it is safe to unwrap because the channels is not empty
+        let channel = channels.channels.first().unwrap();
+        request.channel_id = Some(channel.channel_id.clone());
+        // ctx.channel_manager().close_channel(request);
+        request
+    } else {
+        // No channels with the given peer.
+        return Err(Error::Rpc(RpcError {
+            code: -1,
+            message: format!("No channels with associated peer"),
+            data: None,
+        }));
+    };
+    let resp = ctx.channel_manager().close_channel(res);
+    let (message, channel_id, node_id, funding_utxo) = loop {
```

Comment:
This can create a loop if ldk return an error, so we should check if the is an error

```suggestion
    // FIXME: we should really support the `?` for RPC errors.
    match ctx.channel_manager().close_channel(res) {
        Err(..) => return an error,
        _ => {},
    }
    let (message, channel_id, node_id, funding_utxo) = loop {
```

Reply:
Thanks for catching it!

Reply:
Making an issue for the '?' support.

Reply:
Thanks!